xalia/Win32/HwndListViewProvider.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Xalia.Gudl;
+using Xalia.UiDom;
+using static Xalia.Interop.Win32;
+
+namespace Xalia.Win32
+{
+    internal class HwndListViewProvider : HwndItemListProvider, IWin32Styles
+    {
+        internal HwndListViewProvider(HwndProvider hwndProvider) : base(hwndProvider) { }
+
+        static string[] style_names =
+        {
+            "nosortheader",
+            "nocolumnheader",
+            "noscroll",
+            "ownerdata",
+            "alignleft",
+            "ownerdrawfixed",
+            "editlabels",
+            "autoarrange",
+            "nolabelwrap",
+            "shareimagelists",
+            "sortdescending",
+            "sortascending",
+            "showselalways",
+            "singlesel"
+        };
+
+        static Dictionary<string,int> style_flags;
+
+        static HwndListViewProvider()
+        {
+            style_flags = new Dictionary<string, int>();
+            for (int i=0; i<style_names.Length; i++)
+            {
+                if (style_names[i] is null)
+                    continue;
+                style_flags[style_names[i]] = 0x8000 >> i;
+            }
+        }
+
+        public void GetStyleNames(int style, List<string> names)
+        {
+            switch (style & LVS_TYPEMASK)
+            {
+                case LVS_ICON:
+                    names.Add("icon");
+                    break;
+                case LVS_REPORT:
+                    names.Add("report");
+                    break;
+                case LVS_SMALLICON:
+                    names.Add("smallicon");
+                    break;
+                case LVS_LIST:
+                    names.Add("list");
+                    break;
+            }
+            if ((style & LVS_ALIGNMASK) == 0)
+                names.Add("aligntop");

```

Comment:
Minor nit, but on line 91 of this diff you do:
```
(HwndProvider.Style & LVS_ALIGNMASK) == LVS_ALIGNTOP
```
Might be worth checking against `LVS_ALIGNTOP` here instead of 0 for consistency, although the next line here is already clear I guess.